autogen/oai/client.py

```diff
@@ -345,15 +345,27 @@ def __init__(self, *, config_list: Optional[List[Dict[str, Any]]] = None, **base
                 and additional kwargs.
         """
         openai_config, extra_kwargs = self._separate_openai_config(base_config)
+        # This *may* work if the `llm_config` has specified the `model` attribute,
+        # so just warn here.
         if type(config_list) is list and len(config_list) == 0:
-            logger.warning("openai client was provided with an empty config_list, which may not be intended.")
+            logger.warning("OpenAI client was provided with an empty config_list, which may not be intended.")
+            # If the `llm_config` has no `model` then the call will fail. Abort now.
+            if "model" not in extra_kwargs:
+                raise ValueError("Please specify a value for the 'model' in 'llm_config'.")
 
         self._clients: List[ModelClient] = []
         self._config_list: List[Dict[str, Any]] = []
 
         if config_list:
             config_list = [config.copy() for config in config_list]  # make a copy before modifying
             for config in config_list:
+                # We require that each element of `config_list` has a non-empty value
+                # for `model` specified.
+                model = None
+                if "model" in config:
+                    model = config["model"]
+                if model is None or len(model) == 0:
+                    raise ValueError("Please specify a value for the 'model' in 'config_list'.")
```

Comment:
```suggestion
                    raise ValueError("Please specify a 'model' value for every item in 'config_list'.")
```

Reply:
@ekzhu : Thank you, will update the error message and a corresponding test case.